src/fhir.js

```diff
@@ -5,6 +5,20 @@ const FHIRv300XML = require('./modelInfos/fhir-modelinfo-3.0.0.xml.js');
 const FHIRv400XML = require('./modelInfos/fhir-modelinfo-4.0.0.xml.js');
 const FHIRv401XML = require('./modelInfos/fhir-modelinfo-4.0.1.xml.js');
 
+const memoize = (fn) => {
+    let cache = new Map();
+    return (...args) => {
+       let cacheKey = JSON.stringify([...args]);
```

Comment:
This key can likely get quite large since it's stringifying the passed in `data`, `typeSpecifier`, `modelInfo`, and `suffix`.

In particular, I thought `modelInfo` was going to be a big problem because the `modelInfo` is quite large with several duplicative maps in it -- but running it through the debugger, I see that `JSON.stringify` converts all those huge Maps into empty objects (`{ }`), which turns out to be helpful for us. Still, `modelInfo` should be static and identifiable by its `url` and `version` -- so we could update the key to only use that instead of the whole `modelInfo`. Of course then this is no longer a fully generic memoization function, as it would need to be type-aware.

I'm not sure how to get around the problem when `data` is huge though (for example, if this were being called on a FHIR Bundle with hundreds of embedded resources). In theory, if the `data` is a resource, then the `resourceType`/`id` combination would usually be unique, but there could be extreme edge cases where maybe that's not the case.

Might it make sense to hash the big string and use the hash as the key? I don't remember how expensive a hash operation is though, so I'm not sure how to compare efficiency of having to store very large string keys in the map vs having to hash those strings on every call. I bet @birick1 knows.